core/transactions/transactions_cleanup.cxx

```diff
@@ -273,135 +274,127 @@ transactions_cleanup::create_client_record(const couchbase::transactions::transa
-              details.cas_now_nanos = res.cas;
-              CB_LOST_ATTEMPT_CLEANUP_LOG_DEBUG("get_active_clients found {}", details);
-              return details;
-          } catch (const client_error& e) {
-              auto ec = e.ec();
-              switch (ec) {
-                  case FAIL_DOC_NOT_FOUND:
-                      CB_LOST_ATTEMPT_CLEANUP_LOG_DEBUG("client record not found, creating new one");
-                      create_client_record(keyspace);
-                      throw retry_operation("Client record didn't exist. Creating and retrying");
-                  default:
-                      throw;
-              }
+    client_record_details details;
+    // Write our client record, return details.
+    try {
+        auto id = document_id{ keyspace.bucket, keyspace.scope, keyspace.collection, CLIENT_RECORD_DOC_ID };
+        core::operations::lookup_in_request req{ id };
+        req.specs =
+          lookup_in_specs{
+              lookup_in_specs::get(FIELD_RECORDS).xattr(),
+              lookup_in_specs::get(subdoc::lookup_in_macro::vbucket).xattr(),
           }
-      });
+            .specs();
+        wrap_request(req, config_);
+        auto barrier = std::make_shared<std::promise<result>>();
+        auto f = barrier->get_future();
+        auto ec = config_.cleanup_hooks->client_record_before_get(keyspace.bucket);
+        if (ec) {
+            throw client_error(*ec, "client_record_before_get hook raised error");
+        }
+        cluster_->execute(
+          req, [barrier](core::operations::lookup_in_response resp) { barrier->set_value(result::create_from_subdoc_response(resp)); });
+        auto res = wrap_operation_future(f);
+
+        std::vector<std::string> active_client_uids;
+        auto hlc = res.values[1].content_as();
+        auto now_ms = now_ns_from_vbucket(hlc) / 1000000;
+        details.override_enabled = false;
+        details.override_expires = 0;
+        if (res.values[0].status == subdoc_result::status_type::success) {
+            auto records = res.values[0].content_as();
+            CB_LOST_ATTEMPT_CLEANUP_LOG_TRACE("client records: {}", core::utils::json::generate(records));
+            for (const auto& [key, value] : records.get_object()) {
+                if (key == FIELD_OVERRIDE) {
+                    for (const auto& [override, param] : value.get_object()) {
+                        if (override == FIELD_OVERRIDE_ENABLED) {
+                            details.override_enabled = param.get_boolean();
+                        } else if (override == FIELD_OVERRIDE_EXPIRES) {
+                            details.override_expires = param.as<std::uint64_t>();
+                        }
+                    }
+                } else if (key == FIELD_CLIENTS_ONLY) {
+                    for (const auto& [other_client_uuid, cl] : value.get_object()) {
+                        uint64_t heartbeat_ms = parse_mutation_cas(cl.at(FIELD_HEARTBEAT).get_string());
```

Comment:
could you use the namespace for `uint64_t` and `int64_t` below to ensure that stdlib versions will be used

Reply:
ooh - good catch.   I'll sweep through and make it consistent.